### PR TITLE
[IMP] im_livechat: remove event 'post_message_chat_window' in public livechat code

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2964384

### CODE REVIEW

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js [hunk 1]
```diff
@@ -179,8 +179,13 @@ const PublicLivechatWindow = Widget.extend({
      * @private
      * @param {Object} messageData
      */
-    _postMessage(messageData) {
-        this.trigger_up('post_message_chat_window', { messageData });
+    async _postMessage(messageData) {
+        try {
+            await this.messaging.publicLivechatGlobal.livechatButtonView.sendMessage(messageData);
+        } catch (reason) {
+            reason.event.preventDefault();
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:186]: `reason.event`❓️
`Error.prototype.event` doesn't seem to exist; where is this property supposed to come from?

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js [hunk 2]
```diff
@@ -179,8 +179,13 @@ const PublicLivechatWindow = Widget.extend({
...
+            return this.messaging.publicLivechatGlobal.livechatButtonView.sendMessage(messageData); // try again just in case
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:187]: What if it succeeds the second time? You return whereas you don't do it when it works the first time. This asymmetry of behaviour seems a bit odd to me, I feel like there is something wrong.

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js (hunk 1 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:186]: No idea, good be worth just removing this LOC

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js (hunk 2 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:187]: should have been `await` too, change done

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js [hunk 3]
```diff
@@ -179,8 +179,12 @@ const PublicLivechatWindow = Widget.extend({
      * @private
      * @param {Object} messageData
      */
-    _postMessage(messageData) {
-        this.trigger_up('post_message_chat_window', { messageData });
+    async _postMessage(messageData) {
+        try {
+            await this.messaging.publicLivechatGlobal.livechatButtonView.sendMessage(messageData);
+        } catch (reason) {
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:185]: as you no longer user `reason`, you must add an `_` in front of it or you'll upset the linter 👺

**Review by oomsveta — CHANGES_REQUESTED** [review]

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js (hunk 3 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js]: done

**Review by oomsveta — APPROVED** [review]

**Pull request closed without merging**